http-filter-example/http_filter.cc

```diff
@@ -1,18 +1,60 @@
 #include <string>
+#include <unordered_map>
+#include <sstream>
+#include <vector>
 
 #include "http_filter.h"
 
+#include "source/common/common/logger.h"
 #include "envoy/server/filter_config.h"
 
 namespace Envoy {
 namespace Http {
 
 HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
     const sample::Decoder& proto_config)
-    : key_(proto_config.key()), val_(proto_config.val()) {}
+    : key_(proto_config.key()), val_(proto_config.val()), extra_(stoi(proto_config.extra())) {}
 
 HttpSampleDecoderFilter::HttpSampleDecoderFilter(HttpSampleDecoderFilterConfigSharedPtr config)
-    : config_(config) {}
+    : config_(config) {
+  // populate list of valid operations, implementation will be changed in the next few commits
+  operations_.insert("set-header");
+  
+  const std::string header_config = headerValue();
+
+  // find the position of the first space character
+  const size_t spacePos = header_config.find(' ');
+
+  const auto operation = header_config.substr(0, spacePos);
```

Comment:
BTW, what I said earlier about this being a `string_view` was only half right: it looks like Envoy is using C++14, which doesn't define `std::string_view`, and they're using `absl::string_view` to get views.

I'm not sure if `auto` is smart enough to get the `absl::string_view`. You could verify by removing the `const` and calling `operation = 42;`, which will force a compile failure, and the compiler error should tell you "int not assignable to <type>".

I'm guessing it's not, so you might have to `const absl::string_view operation =` to get the view and avoid the copy.

But again, not a huge deal, especially if it's not in the request path.

And I believe this will get a std::string_view, whenever Envoy upgrades to C++17.

Reply:
It looks like defining operation as a `absl::string_view` prevents me from using `find()` to search for the string_view within `std::unordered_set<std::string> operations_` (the list of valid operations). Will look into this after addressing the comment above about defining magic strings!